frontend/app/components/redesign/components/LinkTagGenerator.tsx

```diff
@@ -0,0 +1,153 @@
+
+    if (validatedPointer) {
+      setParsedLinkTag(parsePointer(validatedPointer))
+      setShowCodeBox(true)
+    } else {
+      setShowCodeBox(false)
+    }
+  }
+
+  const handleOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setPointerInput(e.target.value)
+    setInvalidUrl(false)
+    setShowCodeBox(false)
+  }
+
+  return (
+    <div className="flex w-full max-w-[800px] p-md flex-col gap-md rounded-sm bg-interface-bg-container">
+      {/* Generator Header */}
+      <div>
+        <Heading5>Link tag generator</Heading5>
+      </div>
+      {/* Input Field */}
+      <div>
+        <label
+          htmlFor="paymentPointer"
+          className={cx(
+            'text-field-helpertext-default font-sans text-xs font-normal leading-xs',
+            invalidUrl && 'text-text-error'
+          )}
+        >
+          Your payment pointer
+        </label>
+        <InputField
+          id="paymentPointer"
+          placeholder="Fill in your payment pointer/wallet address"
+          value={pointerInput}
+          onChange={(e) => handleOnChange(e)}
+          error={
+            invalidUrl
+              ? 'The payment pointer you have entered is not valid or cannot be found'
+              : ''
+          }
+        />
+        {!invalidUrl && (
+          <p className="text-xs text-text-secondary mt-1 sm:hidden">
+            Fill in your payment pointer/wallet address
+          </p>
+        )}
+      </div>
+
+      {showCodeBox && linkTag && (
+        <div className="flex min-h-[40px] p-sm justify-between items-center rounded-sm bg-interface-bg-main">
+          <div className="grow shrink font-sans text-sm font-normal leading-normal whitespace-pre-wrap min-w-0 overflow-x-auto">
+            <span>&lt;</span>
+            <span className="text-[#00009F]">link </span>
+            <span className="text-[#00A4DB]">rel</span>
+            <span>=&quot;</span>
+            <span className="text-[#E3116C]">monetization</span>
+            <span>&quot; </span>
+            <span className="text-[#00A4DB]">href</span>
+            <span>=&quot;</span>
+            <span className="text-[#E3116C]">{linkTag}</span>
+            <span>&quot; /&gt;</span>
+          </div>
+          <button
+            onClick={handleCopyClick}
+            aria-label={isCopied ? 'Copied' : 'Copy code to clipboard'}
+          >
+            <SVGCopyIcon />
+          </button>
+        </div>
+      )}
+
+      <ToolsPrimaryButton
+        icon="link"
+        className="justify-center"
+        type="submit"
+        onClick={(e) => handleSubmit(e)}
+      >
+        Generate link-tag
+      </ToolsPrimaryButton>
+
+      {isCopied && (
+        <div className="h-[40px] p-sm rounded-sm bg-interface-bg-main">
+          <p className="font-sans text-sm font-normal leading-normal text-text-success text-center">
+            Copied to clipboard.
+          </p>
+        </div>
+      )}
```

Comment:
Can we (also?) change the icon to a checkbox to show text was copied? The text below can be missed on a smaller screen (think landscape mode).